ubcpi/serialize.py

```diff
@@ -242,7 +244,103 @@ def parse_from_xml(root):
         'options': options,
         'rationale_size': {'min': rationale_size_min, 'max': rationale_size_max},
         'correct_answer': correct_answer,
-        'correct_rationale': {'text': correct_rationale},
+        'correct_rationale': correct_rationale,
         'seeds': seeds,
         'algo': {"name": algo, 'num_responses': num_responses}
     }
+
+
+def serialize_options(options, block):
+    """
+    Serialize the options in peer instruction XBlock to xml
+
+    Args:
+        options (lxml.etree.Element): The <options> XML element.
+        block (PeerInstructionXBlock): The XBlock with configuration to serialize.
+
+    Returns:
+        None
+    """
+    for index, option_dict in enumerate(block.options):
+        option = etree.SubElement(options, 'option')
+        # set correct option and rationale
+        if index == block.correct_answer:
+            option.set('correct', u'True')
+
+            if hasattr(block, 'correct_rationale'):
+                rationale = etree.SubElement(option, 'rationale')
+                rationale.text = unicode(block.correct_rationale['text'])
+
+        text = etree.SubElement(option, 'text')
+        text.text = unicode(option_dict.get('text', ''))
+
+        serialize_image(option, option_dict)
+
+
+def serialize_image(root, image_dict):
+    if 'image_url' not in image_dict:
+        return
+    image = etree.SubElement(root, 'image')
+    image.text = unicode(image_dict.get('image_url', ''))
+    for attr in ['image_position', 'image_show_fields', 'image_alt']:
+        if image_dict.get(attr) is not None:
+            image.set(attr[6:], unicode(image_dict.get(attr)))
+
+
+def serialize_seeds(seeds, block):
+    """
+    Serialize the seeds in peer instruction XBlock to xml
+
+    Args:
+        seeds (lxml.etree.Element): The <seeds> XML element.
+        block (PeerInstructionXBlock): The XBlock with configuration to serialize.
+
+    Returns:
+        None
+    """
+    for index, seed_dict in enumerate(block.seeds):
+        seed = etree.SubElement(seeds, 'seed')
+        # options in xml starts with 1
+        seed.set('option', unicode(seed_dict.get('answer', 0) + 1))
+        seed.text = unicode(seed_dict.get('rationale', ''))
+
+
+def serialize_to_xml(block, root):
+    """
+    Serialize the Peer Instruction XBlock's content to XML.
+
+    Args:
+        block (PeerInstructionXBlock): The peer instruction block to serialize.
+        root (etree.Element): The XML root node to update.
+
+    Returns:
+        etree.Element
+
+    """
+    root.tag = 'ubcpi'
+
+    if block.rationale_size is not None:
+        if block.rationale_size.get('min'):
+            root.set('rationale_size_min', unicode(block.rationale_size.get('min', 1)))
```

Comment:
There's no need to default 'min' to 1 with .get(), since you've already checked it in the previous line.  Or is there something wrong with the logic here?